### PR TITLE
Added new version of wasmtime v0.30.0

### PR DESCRIPTION
**Pull request by redradist**
Specify library name and version:  **wasmtime/0.30.0**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by madebr** [comment]
Shared Windows builds are possible now.
The import library is available since https://github.com/bytecodealliance/wasmtime/issues/3168

**Comment by redradist** [comment]
@madebr 
> 
> 
> Shared mingw builds are possible now. The import library is available since [bytecodealliance/wasmtime#3168](https://github.com/bytecodealliance/wasmtime/issues/3168)

Yes, but it is not possible to remove workaround without removing support for v0.29.0 version, because on v0.29.0 it will fail for mingw

**Comment by madebr** [comment]
> @madebr
> 
> > Shared mingw builds are possible now. The import library is available since [bytecodealliance/wasmtime#3168](https://github.com/bytecodealliance/wasmtime/issues/3168)
> 
> Yes, but it is not possible to remove workaround without removing support for v0.29.0 version, because on v0.29.0 it will fail for mingw

Use `tools.Version` to only raise this error for over versions.

**Comment by redradist** [comment]
@madebr 
> 
> 
> > @madebr
> > > Shared mingw builds are possible now. The import library is available since [bytecodealliance/wasmtime#3168](https://github.com/bytecodealliance/wasmtime/issues/3168)
> > 
> > 
> > Yes, but it is not possible to remove workaround without removing support for v0.29.0 version, because on v0.29.0 it will fail for mingw
> 
> Use `tools.Version` to only raise this error for over versions.

Done, lets see how it will pass CI

File: recipes/wasmtime/all/conanfile.py [hunk 1]
```diff
@@ -69,9 +69,10 @@ def validate(self):
         except KeyError:
             raise ConanInvalidConfiguration("Binaries for this combination of architecture/version/os are not available")
 
-        if (self.settings.compiler, self.settings.os) == ("gcc", "Windows") and self.options.shared:
-            # FIXME: https://github.com/bytecodealliance/wasmtime/issues/3168
-            raise ConanInvalidConfiguration("Shared mingw is currently not possible")
+        if tools.Version(self.version) <= "0.29.0":
+            if (self.settings.compiler, self.settings.os) == ("gcc", "Windows") and self.options.shared:
+                # FIXME: https://github.com/bytecodealliance/wasmtime/issues/3168
```
Comment (madebr) [inline recipes/wasmtime/all/conanfile.py:74]:
```suggestion
                # https://github.com/bytecodealliance/wasmtime/issues/3168
```

**Review by madebr — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
🚀Amazing work contributing back to OSS ❤️

**Review by SSE4 — APPROVED** [review]

**Pull request merged**